aio/tools/transforms/angular-base-package/post-processors/auto-link-code.js

```diff
@@ -30,51 +30,64 @@ module.exports = function autoLinkCode(getDocFromAlias) {
-                const nodes =
-                    textContent(node)
-                        .split(/([A-Za-z0-9_.-]+)/)
-                        .filter(word => word.length)
-                        .map((word, index, words) => {
-                          // remove docs that fail the custom filter tests
-                          const filteredDocs = autoLinkCodeImpl.customFilters.reduce(
-                              (docs, filter) => filter(docs, words, index), getDocFromAlias(word));
-                          return foundValidDoc(filteredDocs, word, file) ?
-                              // Create a link wrapping the text node.
-                              createLinkNode(filteredDocs[0], word) :
-                              // this is just text so push a new text node
-                              {type: 'text', value: word};
-                        });
+        visit(node, 'text', (node, ancestors) => {
+          // Only interested in text nodes that are not inside links
+          const isInsideLink = ancestors.every(ancestor => !is(ancestor, 'a'));
+          if (!isInsideLink) {
+            return;
+          }
 
-                // Replace the text node with the links and leftover text nodes
-                Array.prototype.splice.apply(parent.children, [index, 1].concat(nodes));
-              }
-            }
-          });
-        }
+          const parent = ancestors[ancestors.length - 1];
+          const index = parent.children.indexOf(node);
+
+          // Can we convert the whole text node into a doc link?
+          const docs = getDocFromAlias(node.value);
+          if (foundValidDoc(docs, node.value, file)) {
+            parent.children.splice(index, 1, createLinkNode(docs[0], node.value));
+          } else {
+            // Parse the text for words that we can convert to links
+            const nodes = getNodes(node, file);
+            // Replace the text node with the links and leftover text nodes
+            Array.prototype.splice.apply(parent.children, [index, 1].concat(nodes));
+          }
+        });
       });
     };
   }
 
+  function isValidCodeElement(node, ancestors) {
+    // Only interested in code elements that:
+    // * do not have `no-auto-link` class
+    // * do not have an ignored language
+    // * are not inside links
+    const containsElement = autoLinkCodeImpl.codeElements.some(elementType => is(node, elementType));
+    const hasNoAutoLink = !node.properties.className || !node.properties.className.includes('no-auto-link');
+    const isLanguageSupported = !autoLinkCodeImpl.ignoredLanguages.includes(node.properties.language);
+    const isInsideLink = ancestors.every(ancestor => !is(ancestor, 'a'));
+    return containsElement && hasNoAutoLink && isLanguageSupported && isInsideLink;
+  }
+
+  function getNodes(node, file) {
+    return textContent(node)
+      .split(/([A-Za-z0-9_.-]+)/)
+      .filter(word => word.length)
```

Comment:
While you are at it, let's do one more improvement:
The current implementation will also consider whitespace-only tokens. For example, `"foo bar"` will result into 3 words - "foo", " " and "bar". Looking for docs to match whitespace-only nodes is unnecessary. Let's fix it with:

```suggestion
      .filter(word => word.trim().length > 0)
```

Reply:
I tried that, but it seems we need the spaces. Without that, the code is converted to:
```typescript
abstractclassChangeDetectorRef {
  abstract markForCheck(): voidabstract detach(): voidabstract detectChanges(): voidabstract checkNoChanges(): voidabstract reattach(): void
}
```
vs

```typescript
abstract class ChangeDetectorRef {
  abstract markForCheck(): void
  abstract detach(): void
  abstract detectChanges(): void
  abstract checkNoChanges(): void
  abstract reattach(): void
}
```

Reply:
Did you try my proposed change exaclty? It is not possible for my changes to have changed the resulting HTML (since strings are immutable and we only use `.trim()` in `filter` (which is not mutating the array)).

Reply:
I did. It's not working for me as expected.

Reply:
Oh, I see - I didn't realize we are using these nodes to replace the original code element. We need the whitespace then, yes.

Let's leave it as is then. Thx for trying it out and sorry for the noise 😅